packages/test/test-utils/src/testObjectProvider.ts

```diff
@@ -185,9 +185,15 @@ export class EventAndErrorTrackingLogger extends TelemetryLogger {
         const unexpectedErrors = this.unexpectedErrors.splice(0, this.unexpectedErrors.length);
         return {
             expectedNotFound,
-            unexpectedErrors,
+            unexpectedErrors: this.excludeAllowedErrors(unexpectedErrors),
         };
     }
+
+    private excludeAllowedErrors(errors: ITelemetryBaseEvent[]) {
+        return errors.filter((event) => !(event.eventName in [
+            "NoRealStorageInDetachedContainer", // Removed in current version, but still present in previous versions
```

Comment:
This is not ideal.  But there are hundreds of failures hitting this so I don't think it's reasonable to update all tests to expect this.

To clarify my comment in the code here, this error event does not exist in main anymore.  But it does in the old compat versions.  It was removed because it's harmless and unhelpful (the code throws, callers can decide if it's a problem or not).

@anthony-murphy - Any tips or feedback on this?  I do have an alternate approach that avoid the codepath hitting this error altogether, but it's not as correct in terms of encapsulation of layers IMO.

Reply:
(I see that `x in array` doesn't work like I thought. Updated in later iteration)